src/darwin/Framework/CHIP/CHIPThreadOperationalDataset.mm

```diff
@@ -0,0 +1,94 @@
+/**
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "CHIPThreadOperationalDataset.h"
+
+#include "CHIPLogging.h"
+#include <support/Span.h>
+#include <support/ThreadOperationalDataset.h>
+
+size_t const CHIPSizeThreadNetworkName = chip::Thread::kSizeNetworkName;
+size_t const CHIPSizeThreadExtendedPanId = chip::Thread::kSizeExtendedPanId;
+size_t const CHIPSizeThreadMasterKey = chip::Thread::kSizeMasterKey;
+size_t const CHIPSizeThreadMeshLocalPrefix = chip::Thread::kSizeMeshLocalPrefix;
+size_t const CHIPSizeThreadPSKc = chip::Thread::kSizePSKc;
+
+@interface CHIPThreadOperationalDataset ()
+
+@property (readonly) chip::Thread::OperationalDataset cppThreadOperationalDataset;
+
+@end
+
+@implementation CHIPThreadOperationalDataset
+
+- (nullable instancetype)initWithNetworkName:(NSString *)networkName
+                               extendedPANID:(NSData *)extendedPANID
+                                   masterKey:(NSData *)masterKey
+                                        PSKc:(NSData *)PSKc
+                                     channel:(uint16_t)channel
+                                       panID:(NSData *)panID
+{
+    if (self = [super init]) {
+        _networkName = networkName;
+        _extendedPANID = extendedPANID;
+        _masterKey = masterKey;
+        _PSKc = PSKc;
+        _channel = channel;
+        _panID = panID;
+        _cppThreadOperationalDataset = chip::Thread::OperationalDataset();
+        [self _populateCppOperationalDataset];
+        if (!_cppThreadOperationalDataset.IsValid(_cppThreadOperationalDataset.AsByteSpan())) {
```

Comment:
This `IsValid` check doesn't really make sense.  All it validates is that the raw bytes are made up of `ThreadTLV` objects, which will always be true if the span came from an `AsByteSpan` call on a `chip::Thread::OperationalDataset`.  Just take this part out?

Reply:
ah, unfortunate. Okay I'll remove it.